src/ray/raylet/node_manager.cc

```diff
@@ -993,6 +994,9 @@ void NodeManager::ProcessWaitRequestMessage(
                                  fbb.GetSize(), fbb.GetBufferPointer());
         if (status.ok()) {
           // The client is unblocked now because the wait call has returned.
+          for (const auto &object_id : found) {
+            task_dependency_manager_.UnsubscribeDependency(current_task_id, object_id);
```

Comment:
Instead of having this method be public, I think we can call this whenever the `object_id` becomes local in the `TaskDependencyManager`.